habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -386,6 +386,290 @@ def step(self, *args, is_last_action, **kwargs):
             return {}
 
 
+class Controller:
+    """
+        A controller that takes the input of the waypoints and
+        produces the velocity command.
+    """
+    def __init__(self, v_max_default, w_max_default, track_yaw=True):
+        # If we want track yaw or not
+        self.track_yaw = track_yaw
+
+        # Parameter of the controller
+        self.v_max = v_max_default
+        self.w_max = w_max_default
+
+        self.lin_error_tol = self.v_max / 120
+        self.ang_error_tol = self.w_max / 120
+
+        self.acc_lin = 2.4
+        self.acc_ang = 2.4
+
+        # Initialize the parameters
+        self.base_pose_goal = np.zeros(3)
+        self.base_velocity_goal = np.zeros(3)
+
+    def set_goal(self, goal, vel_goal=None):
+        """
+        Set the goal of the agent
+        """
+        self.base_pose_goal = goal
+        if vel_goal is not None:
```

Comment:
if this is None, should it be reset to zeros? what's the expected behavior here?

Reply:
Thanks for this! I think the vel_goal will be zero in the end if it is none